File: android/app/src/main/kotlin/com/rkfl/flutter_music_player/models/AudioModel.kt
package com.rkfl.flutter_music_player.models

data class AudioModel(
        val id: Int,
        val path: String,
        val name: String,
        val album: String,
        val artist: String,
        val playbackDuration: Int
) {

    override fun toString(): String {
        val sb = StringBuffer()

        sb.append("{")
        sb.append("\"id\": ").append(this.id).append(",")
        sb.append("\"path\": \"").append(this.path).append("\",")
        sb.append("\"name\": \"").append(this.name).append("\",")
        sb.append("\"album\": \"").append(this.album).append("\",")
        sb.append("\"artist\": \"").append(this.artist).append("\",")
        sb.append("\"playbackDuration\": ").append(this.playbackDuration)
        sb.append("}")

        return sb.toString()
    }

}

File: lib/models/AudioModel.dart
import 'dart:convert';

class AudioModel {

  final int id;
  final String path;
  final String name;
  final String album;
  final String artist;
  final double playbackDuration;

  AudioModel([
      this.id = 0,
      this.path = "",
      this.name = "",
      this.album = "",
      this.artist = "",
      this.playbackDuration = 0.0
  ]);

  static AudioModel fromJson(dynamic json) {
    dynamic jsonTmp = jsonDecode(json);

    return AudioModel(
        jsonTmp['id'],
        jsonTmp['path'],
        jsonTmp['name'],
        jsonTmp['album'],
        jsonTmp['artist'],
        jsonTmp['playbackDuration'].toDouble()
    );
  }

}